components/org.wso2.carbon.identity.api.server.branding.preference.management/org.wso2.carbon.identity.api.server.branding.preference.management.v1/src/main/resources/branding-preference.yaml

```diff
@@ -0,0 +1,278 @@
+            * /permission/admin/manage/identity/applicationmgt/update <br>
+          <b>Scope required:</b> <br>
+            * internal_application_mgt_update
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/BrandingPreferenceModel'
+        description: This represents the branding preferences to be updated.
+        required: true
+      responses:
+        '200' :
+          description: Successfully updated
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+    delete:
+      tags:
+        - Branding Preference
+      operationId: deleteBrandingPreference
+      summary: Deletes branding preferences of Tenant.
+      description: |
+        This API provides the capability to delete the branding preferences of the tenant.<br>
+          <b>Permission required:</b> <br>
+            * /permission/admin/manage/identity/applicationmgt/update <br>
+          <b>Scope required:</b> <br>
+            * internal_application_mgt_update
+      parameters:
+        - $ref: '#/components/parameters/typeQueryParam'
+        - $ref: '#/components/parameters/nameQueryParam'
+        - $ref: '#/components/parameters/localeQueryParam'
+      responses:
+        '204' :
+          description: Successfully deleted.
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
+          $ref: '#/components/responses/Forbidden'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+components:
+  parameters:
+    typeQueryParam:
+      in: query
+      name: type
+      required: false
+      description: Resource Type to filter the retrieval of themes.
+      schema:
+        type: string
+        enum:
+          - ORG
+          - APP
+          - CUSTOM
+      example: "ORG"
+    nameQueryParam:
+      in: query
+      name: name
+      required: false
+      description: Application name/ Tenant name to filter the retrieval of themes.
+      schema:
+        type: string
+      example: "NAME"
+    localeQueryParam:
+      in: query
+      name: locale
+      required: false
+      description: Resource Localye to filter the retrieval of themes.
+      schema:
+        type: string
+      example: "en-US"
+
+  schemas:
+    BrandingPreferenceModel:
+      required:
+        - type
+        - object
+      type: object
+      properties:
+        type:
+          type: string
+          enum:
+            - ORG
+            - APP
+            - CUSTOM
+          example: "ORG"
+        name:
```

Comment:
Isn't name is a mandatory/required attribute here ?

Reply:
No akka. With the current implementation we are only supported for the tenant wise branding, In that case we can retrieve the tenant-domain path param. In future phases for apllication wise theming we can mandate name from the backend level.